recipes-framework/tensorflow-lite/libtensorflow-lite-c_2.16.1.bb

```diff
@@ -41,34 +41,41 @@ OECMAKE_SOURCEPATH = "${S}/tensorflow/lite/c"
 # XNNPack is valid only on 64bit. 
 # In the case of arm 32bit, it will be turned off because the build will be
 # an error depending on the combination of target CPUs.
+EXTRA_OECMAKE:append:raspberrypi0 = " -DTFLITE_ENABLE_XNNPACK=OFF"
+EXTRA_OECMAKE:append:raspberrypi0-wifi = " -DTFLITE_ENABLE_XNNPACK=OFF"
+EXTRA_OECMAKE:append:raspberrypi0-2w-64 = " -DTFLITE_ENABLE_XNNPACK=ON"
+EXTRA_OECMAKE:append:raspberrypi-cm = " -DTFLITE_ENABLE_XNNPACK=OFF"
+EXTRA_OECMAKE:append:raspberrypi-cm3 = " -DTFLITE_ENABLE_XNNPACK=OFF"
+EXTRA_OECMAKE:append:raspberrypi = " -DTFLITE_ENABLE_XNNPACK=OFF"
+EXTRA_OECMAKE:append:raspberrypi2 = " -DTFLITE_ENABLE_XNNPACK=OFF"
+EXTRA_OECMAKE:append:raspberrypi3 = " -DTFLITE_ENABLE_XNNPACK=OFF"
+EXTRA_OECMAKE:append:raspberrypi3-64 = " -DTFLITE_ENABLE_XNNPACK=ON"
+EXTRA_OECMAKE:append:raspberrypi4 = " -DTFLITE_ENABLE_XNNPACK=OFF"
+EXTRA_OECMAKE:append:raspberrypi4-64 = " -DTFLITE_ENABLE_XNNPACK=ON"
+EXTRA_OECMAKE:append:raspberrypi5 = " -DTFLITE_ENABLE_XNNPACK=ON"
```

Comment:
You are doing configuration per Machine, it would be better to do configuration per Architecture

Here you want to enable for aarch64 but not for armv7
so you could just do

EXTRA_OECMAKE:append:aarch64 = " -DTFLITE_ENABLE_XNNPACK=ON"

Another improvement is to use PackageConfig

Declare a PackageConfig like 

PACKAGECONFIG[xnnpack] = "-DTFLITE_ENABLE_XNNPACK=ON, -DTFLITE_ENABLE_XNNPACK=OFF"

An then set the default PACAKAGECONFIG regarding your architecture

PACKAGECONFIG:append:<arch> = "xnnpack"

Reply:
Change the configuration to EXTRA_OECMAKE by architecture, not by machine.